src/4.3_to_5.0/migrate_43_to_50.ts

```diff
@@ -201,42 +200,39 @@ const accumulateStructure = ({
 };
 
 /**
- * Returns the converted UI folder, ready to be used by ActiveUI 5.
- * Also returns a count of the number of migration successes and failures and an detailed error report.
+ * Modifies the ActiveUI 4.3 `contentServer` structure, for it to be ready to be used by ActiveUI 5.0.
+ * Also keeps the number of migration successes and failures in `counters` and a detailed `errorReport`.
  *
  * Widgets with keys in `keysOfWidgetPluginsToRemove` are not migrated:
- * - for a matching saved ActiveUI 4 widget, no ActiveUI 5 file is created.
- * - for a saved ActiveUI 4 dashboard including a matching widget, the widget is removed from the output ActiveUI 5 dashboard, and the layout is adapted so that siblings take the remaining space.
+ * - for a matching saved ActiveUI 4.3 widget, no ActiveUI 5.0 file is created.
+ * - for a saved ActiveUI 4.3 dashboard including a matching widget, the widget is removed from the output ActiveUI 5.0 dashboard, and the layout is adapted so that siblings take the remaining space.
+ *
+ * Mutates `contentServer`, `errorReport` and `counters`.
  */
 export async function migrate_43_to_50(
-  legacyUIFolder: ContentRecord,
+  contentServer: ContentRecord,
   {
+    errorReport,
+    counters,
```

Comment:
What's the logic in adding these as arguments instead of creating them from scratch?

Reply:
They now represent global variables that might be used throughout successive migrations. For example, consider a 4.3 --> 5.1 migration. We'll run `migrate_43_to_50`, then (the not yet existing) `migrate_50_to_51`. In the end, we want `counters` and `errorReport` to contain information from both migrations.

Hence why we want to define them at a higher level, and populate them (by mutating them) inside, between other migration functions, `migrate_43_to_50`.